pkl-codegen-java/src/main/kotlin/org/pkl/codegen/java/JavaCodeGenerator.kt

```diff
@@ -872,6 +872,15 @@ class JavaCodeGenerator(
       } else key
     }
   }
+
+  private val nameMapper = NameMapper(codegenOptions.renames)
+
+  private fun mapModuleName(name: String): kotlin.Pair<String, String> {
+    val mappedName = nameMapper.map(name)
+    val packageName = mappedName.substringBeforeLast('.', "")
+    val className = mappedName.substringAfterLast('.').replaceFirstChar { it.titlecaseChar() }
+    return packageName to className
+  }
```

Comment:
Now that this is a straight up rename, I think we should adjust this so that we don't then change the name again if it's been explicitly renamed.

I think the logic here can be adjusted to something like:

```kt
  private fun mapModuleName(name: String): kotlin.Pair<String, String> {
    val (mappedName, isClassRenamed) = nameMapper.map(name)
    val packageName = mappedName.substringBeforeLast('.', "")
    val className = mappedName.substringAfterLast('.').let { segment ->
      if (isClassRenamed) segment else segment.replaceFirstChar { it.titlecaseChar() }
    }
    return packageName to className
  }
```

Where `org.pkl.commons.NameMapper#map` gets changed to:

```kt
  fun map(sourceName: String): Pair<String, Boolean> {
    for ((sourcePrefix, targetPrefix) in sortedMapping) {
      if (sourceName.startsWith(sourcePrefix)) {
        val remainder = sourceName.substring(sourcePrefix.length)
        return targetPrefix + remainder to ((sourcePrefix.length - 1) > sourceName.lastIndexOf('.'))
      }
    }
    return sourceName to false
  }
```

Reply:
Alternatively, `org.pkl.commons.NameMapper#map` just returns a package name and class name tuple.